### PR TITLE
Adds documentation for the "browser" pattern

### PR DESCRIPTION
**Pull request by kristoferjoseph**
This update documents the workflow the new arc-plugin-rollup plugin enables.

### CODE REVIEW

File: app/docs/md/learn/starter-project/browser.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+<doc-callout level="info" mark="💭">
+
+The `/_public` endpoint is created for you so that Enhance can do the tedious work of replacing your authored file name with a fingerprinted one to avoid caching issues.
+
+</doc-callout>
+
+## Share elements with the browser
+
+Now that you get the basics of the browser bundle workflow let's look at how you might share your elements with the browser for progressive enhancement.
+
```
Comment (macdonst) [inline app/docs/md/learn/starter-project/browser.md:50]: We might want to add a non-element example of sharing code between the server and browser.

File: app/docs/md/learn/starter-project/structure.md [hunk 2]
```diff
@@ -36,10 +38,20 @@ Elements must be [named](https://html.spec.whatwg.org/multipage/custom-elements.
...
 
 </doc-callout>
+
+## Browser
+
+The `browser` folder is preconfigured to output a bundle to be used when progressively enhancing your pages in the browser. Files in `app/browser` are bundled to `/public/pages/`. Meaning `app/browser/index.mjs` will be bundled with any imported dependencies to `/public/pages/index.mjs` and available to be loaded by a script tag at `/_public/pages/index.mjs`.
+
+<doc-callout level="none" mark="🦄">
+
+**[Read more about `browser` routes here →](/docs/learn/starter-project/browser)**
```
Comment (macdonst) [inline app/docs/md/learn/starter-project/structure.md:55]: Should we be using `routes` for this label?

**Review by macdonst — APPROVED** [review]
Looks good to me. One possible addition and one question about wording.

File: app/docs/md/learn/starter-project/browser.md (hunk 1 above)
Comment (kristoferjoseph) [reply app/docs/md/learn/starter-project/browser.md:50]: I'll add it back.

File: app/docs/md/learn/starter-project/structure.md (hunk 2 above)
Comment (kristoferjoseph) [reply app/docs/md/learn/starter-project/structure.md:55]: Good catch.

File: app/docs/md/learn/starter-project/browser.md (hunk 1 above)
Comment (kristoferjoseph) [reply app/docs/md/learn/starter-project/browser.md:50]: Great idea

**Review by tbeseda — APPROVED** [review]
noice

**Pull request merged**